docs/rebase.md

```diff
@@ -0,0 +1,171 @@
+# Rebase on top of latest stable version of kubernetes
+
+- [Overview](#overview)
+- [Prerequisites](#prerequisites)
+- [Rebase process](#rebase-process)
+  1. [Latest kubernetes](#latest-kubernetes)
+  2. [Kubernetes fork](#kubernetes-fork)
+  3. [Cherry picks](#cherry-picks)
+  4. [Origin restore deps](#origin-restore-deps)
+  5. [Checkout kubernetes](#checkout-kubernetes)
+  6. [Kubernetes restore deps](#kubernetes-restore-deps)
+  7. [Update deps](#update-deps)
+  8. [Review changes](#review-changes)
+  9. [Compilation and tests](#compilation-and-tests)
+
+
+
+| NOTE |
+| ---- |
+| Even though all the steps provided in this document apply mostly to kubernetes rebase. Yet still, some parts are easily applicable to other dependencies. |
+
+
+## Overview
+
+In short the rebase process can be described with the following steps:
+
+1. Pick latest stable kubernetes version you want to base your work on.
+2. Create a `release-X.X.X` branch in [our kubernetes fork](https://github.com/openshift/kubernetes/),
+   where `X.X.X` reflects the kubernetes version from step 1 and push its
+   contents into it.
+3. Review *ALL* cherry picks applied to a current version of kubernetes in our fork
+   and apply them accordingly in a new PR against our fork targeting branch
+   created in previous step.
+4. Run `hack/godep-restore.sh` in origin repository to have appropriate state of
+   dependencies.
+5. Inside kubernetes repository checkout the desired kubernetes level with all
+   the necessary cherry picks (iow. result of step 3).
+6. Run `hack/godep-restore.sh` inside kubernetes repository to get the new level
+   of required dependencies.
+7. Run `hack/godep-save.sh` inside origin repository to get the required changes
```

Comment:
this command fails because the origin code base cannot be compiled against a new Kube.

Instead: 
7 a) copy a kube snapshot manually into vendor/, get the origin code to compile (step 9).
   b) hack/godep-save.sh

Reply:
It works just fine. I was using it with success during 1.7.6 rebase, as well as during bumping etcd.

Reply:
1.7.0 -> 1.7.6 is not a major version rebase, usually no packages structure changes. If it works, even better. Just be prepared if it doesn't.

Reply:
We need to make sure this works, at all time. There's too much craziness ;)